src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -140,9 +140,9 @@ class ArrayBuffer[A] private (initialElements: Array[AnyRef], initialSize: Int)
   def addOne(elem: A): this.type = {
     mutationCount += 1
     val newSize = size0 + 1
-    ensureSize(newSize)
+    if(array.length <= newSize - 1) ensureSize(newSize)
```

Comment:
if(array.length <= size0) ensureSize(newSize)

Reply:
This was actually intended to account for the possibility of integer overflow on `size0 + 1`.

With that explanation: keep as written, add source-code comment, or am I wrong here?

Reply:
thanks then better with a comment around :)